smdebug/core/index_reader.py

```diff
@@ -233,28 +235,45 @@ def _update_tensors_from_json(
         mode = index_meta["mode"]
         mode = ModeKeys[mode.strip()]
         mode_step = index_meta["mode_step"]
-        event_file_name = os.path.join(path, index_meta["event_file_name"])
-        tensors = index_dict["tensor_payload"]
-        for tensor in tensors:
+
+        if "event_file_name" in index_meta:
+            event_file_name = os.path.join(path, index_meta["event_file_name"])
+        else:
+            event_file_name = None
+
+        to_update_index_dict = []
+
+        tensor_payload = index_dict.get("tensor_payload", [])
+        for tensor in tensor_payload:
             tensor_name = tensor["tensorname"]
             start_idx = tensor["start_idx"]
             length = tensor["length"]
             tensor_location = TensorLocation(
                 tensor_name, mode, mode_step, event_file_name, start_idx, length, worker
```

Comment:
Can `event_file_name` be `None` here?

`if event_file_name in index_meta is False` and `len(tensor_payload)>0`.

Is the TensorLocation designed to handle this?

Is there a possibility for the user to attempt to access  the tensor value when it is None?

Reply:
Previously the expectation was that event_file_name is always present when tensor_payload is in the index file. But good point it could have led to silent failures or issues further down the line. I've refactored the code. Now if event_file_name is not present when reading tensor_payload, it'll throw KeyError

Reply:
> Is there a possibility for the user to attempt to access the tensor value when it is None but there is corresponding shape payload

That's very much possible and would raise TensorUnavailableForStep exception.